vcr/api/v2/api.go

```diff
@@ -238,14 +239,41 @@ func (w *Wrapper) CreateVP(ctx context.Context, request CreateVPRequestObject) (
 		expires = &parsedTime
 	}
 
-	proofOptions := proof.ProofOptions{
-		Created:   created,
-		Domain:    request.Body.Domain,
-		Challenge: request.Body.Challenge,
-		Expires:   expires,
+	presentationOptions := holder.PresentationOptions{
+		ProofOptions: proof.ProofOptions{
+			Created:   created,
+			Domain:    request.Body.Domain,
+			Challenge: request.Body.Challenge,
+			Expires:   expires,
+		},
+	}
+
+	// custom proofPurpose
+	if request.Body.ProofPurpose != nil {
+		presentationOptions.ProofOptions.ProofPurpose = *request.Body.ProofPurpose
+	}
+
+	// pass context and type as ssi.URI
+	if request.Body.Context != nil {
+		for _, sc := range *request.Body.Context {
```

Comment:
(at least, I think so)

```suggestion
		for sc := range *request.Body.Context {
```

Reply:
That would give numbers and then tests fail